framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -70,6 +72,98 @@ public ValueTransfer(CFAbstractAnalysis<CFValue, CFStore, CFTransfer> analysis)
     }
 
     /**
+     * Returns a range of possible lengths for {@code subNode}, as casted to a String.
+     *
+     * @param stringLengths a list of string lengths for subNode, or null
+     */
+    private Range getStringLengthRange(
+            Node subNode, TransferInput<CFValue, CFStore> p, List<Integer> stringLengths) {
+        if (stringLengths != null) {
+            return ValueCheckerUtils.getRangeFromValues(stringLengths);
+        }
+
+        CFValue value = p.getValueOfSubNode(subNode);
+
+        AnnotationMirror arrayLenRangeAnno =
+                AnnotationUtils.getAnnotationByClass(value.getAnnotations(), ArrayLenRange.class);
+
+        if (arrayLenRangeAnno != null)
+            return ValueAnnotatedTypeFactory.getArrayLenRange(arrayLenRangeAnno);
+
+        // @StringVal, @UnknownVal, @BottomVal
+        AnnotationMirror topAnno =
```

Comment:
For `UnknownVal` and `BottomVal`, you're not using the result of the call to `getAnnotationByClass` so you can just use `AnnotationUtils#hasAnnotation` instead